src/System.Security.Cryptography.Cng/tests/AesCngTests.cs

```diff
@@ -0,0 +1,96 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Security.Cryptography.Cng.Tests
+{
+    public static class AesCngTests
+    {
+        private const int BlockSizeBytes = 16;
+
+        private static readonly CngAlgorithm s_cngAlgorithm = new CngAlgorithm("AES");
+
+        // The ephemeral key has already been validated by the AesCipherTests suite.
+        // Therefore we can use the ephemeral key to validate the persisted key.
+        [OuterLoop]
+        [ConditionalTheory(nameof(SupportsPersistedSymmetricKeys))]
+        // AES128-ECB-NoPadding 2 blocks.
+        [InlineData(128, 2 * BlockSizeBytes, CipherMode.ECB, PaddingMode.None)]
+        // AES128-CBC-NoPadding at 2 blocks
+        [InlineData(128, 2 * BlockSizeBytes, CipherMode.CBC, PaddingMode.None)]
+        // AES256-CBC-Zeros at 1.5 blocks 
+        [InlineData(256, BlockSizeBytes + BlockSizeBytes / 2, CipherMode.CBC, PaddingMode.Zeros)]
+        // AES192-CBC-PKCS7 at 1.5 blocks 
+        [InlineData(192, BlockSizeBytes + BlockSizeBytes / 2, CipherMode.CBC, PaddingMode.PKCS7)]
+        public static void VerifyPersistedKey(
+            int keySize,
+            int plainBytesCount,
+            CipherMode cipherMode,
+            PaddingMode paddingMode)
+        {
+            SymmetricCngTestHelpers.VerifyPersistedKey(
+                s_cngAlgorithm,
+                keySize,
+                plainBytesCount,
+                keyName => new AesCng(keyName),
+                () => new AesCng(),
+                cipherMode,
+                paddingMode);
+            
+        }
+
+        [OuterLoop]
+        [ConditionalFact(nameof(SupportsPersistedSymmetricKeys))]
+        public static void GetKey_NonExportable()
+        {
+            SymmetricCngTestHelpers.GetKey_NonExportable(
+                s_cngAlgorithm,
+                keyName => new AesCng(keyName));
+        }
+
+        [OuterLoop]
+        [ConditionalFact(nameof(SupportsPersistedSymmetricKeys))]
+        public static void SetKey_DetachesFromPersistedKey()
+        {
+            SymmetricCngTestHelpers.SetKey_DetachesFromPersistedKey(
+                s_cngAlgorithm,
+                keyName => new AesCng(keyName));
+        }
+
+        [OuterLoop]
+        [ConditionalFact(nameof(SupportsPersistedSymmetricKeys))]
+        public static void LoadWrongKeyType()
+        {
+            string keyName = Guid.NewGuid().ToString();
+            CngKey cngKey = CngKey.Create(new CngAlgorithm("3DES"), keyName);
+
+            try
+            {
+                Assert.Throws<CryptographicException>(() => new AesCng(keyName));
+            }
+            finally
+            {
+                cngKey.Delete();
+            }
+        }
+
+        [OuterLoop]
+        [ConditionalFact(nameof(SupportsPersistedSymmetricKeys))]
+        public static void VerifyMachineKey()
+        {
+
```

Comment:
Bah, missed this one.  If I have to make any other changes I'll try to hit this one again.